src/main/java/io/appium/java_client/remote/options/W3CCapabilityKeys.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.remote.options;
+
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+public class W3CCapabilityKeys implements Predicate<String> {
+    private static W3CCapabilityKeys instance = null;
+    private static final Predicate<String> ACCEPTED_W3C_PATTERNS = Stream.of(
+                    "^[\\w-]+:.*$",
+                    "^acceptInsecureCerts$",
+                    "^browserName$",
+                    "^browserVersion$",
+                    "^platformName$",
+                    "^pageLoadStrategy$",
+                    "^proxy$",
+                    "^setWindowRect$",
+                    "^strictFileInteractability$",
+                    "^timeouts$",
+                    "^unhandledPromptBehavior$",
+                    "^webSocketUrl$")  // from webdriver-bidi
+            .map(Pattern::compile)
+            .map(Pattern::asPredicate)
+            .reduce(identity -> false, Predicate::or);
+
+    @Override
+    public boolean test(String capabilityName) {
+        return ACCEPTED_W3C_PATTERNS.test(capabilityName);
+    }
+
+    /**
+     * Returns class singleton instance.
+     *
+     * @return Singleton object.
+     */
+    public static synchronized W3CCapabilityKeys getInstance() {
```

Comment:
Also we need to make `W3CCapabilityKeys` constructor `private`

Reply:
I don't mind changing it, but I would like to see some actual investigations/numbers related to the performance degradation issue.

Reply:
`synchronized` at method level leads to the limitation that only one thread can have access to the method at the given time, so in multithreaded context there could be a queue of threads waiting their order to invoke the method (of course possible performance drawback depends on the number of threads).

According to https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java:
> Since synchronizing a method could in some extreme cases decrease performance by a factor of 100 or higher...

I don't know the actual numbers and haven't checked it on my own, but that's kind of classical performance problem solved in Java-world.

Reply:
https://objectpartners.com/2011/11/29/java-synchronization-method-performance-comparison/ 
I believe with the current Java releases it's even less. Nevertheless this particular lib is not intended to be used in high-perf realtime systems, where each millisecond counts.